developer/frameworks/articles/02-dependency-injection/02-declarative-services/99-converting-a-service-builder-module-from-spring-di-to-osgi-ds.markdown

```diff
@@ -0,0 +1,108 @@
+# Converting a Service Builder Module from Spring DI to OSGi DS
+
+Prior to @product@ 7.2, Service Builder modules used Spring for dependency
+injection (DI). Now OSGi Declarative Services (DS) is the default dependency
+injection mechanism for new Service Builder modules. It's easier to learn and
+fosters loose coupling between services. If you have an existing Service Builder
+module that uses Spring DI, you can modify it to use DS. 
+
+Here are the conversion steps:
+
+- Step 1: Prepare your project for DS 
+
+- Step 2: Update your Spring bean classes 
+
+- Step 3: Resolve any circular dependencies 
+
+Now prepare your project. 
+
+## Step 1: Prepare your project for DS 
+
+Setting up your project's metadata, dependencies, and `service.xml` is
+straightforward. 
+
+1.  Enable the DS annotation option for your inherited dependencies by adding 
+    this line to your bnd File:
+
+        -dsannotations-options: inherit
+    
+2.  Since DS Service Builder modules use the AOP API, add it as a compile 
+    dependency: 
+
+        com.liferay:com.liferay.portal.aop.api:1.0.0
+
+3.  Add the `dependency-injector="ds"` attribute to your `service.xml` file's 
+    `<service-builder>` element. 
+
+## Step 2: Update your Spring bean classes
+
+Service Builder regenerates your the standard interfaces automatically. You must update your non-generated Spring bean classes to use DS. Here are the steps: 
+
+1.  If the class implements an interface, add the `@Component` class annotation 
```

Comment:
"For your *ServiceImpls, *LocalServiceImpls and *FinderImpls add the ```@Componet``` annotation to the class."